nilearn/reporting/glm_reporter.py

```diff
@@ -967,3 +973,289 @@ def _dataframe_to_html(df, precision, **kwargs):
+        contrasts,
+        output_type='all',
+    )
+
+    # Model metadata
+    # TODO: Determine optimal mapping of model metadata to BIDS fields.
+    model_metadata_file = os.path.join(
+        out_dir,
+        '{}statmap.json'.format(prefix),
+    )
+    dataset_description_file = os.path.join(
+        out_dir,
+        'dataset_description.json',
+    )
+
+    DATA_ATTRIBUTES = [
+        't_r',
+    ]
+    PARAMETER_ATTRIBUTES = [
+        'drift_model',
+        'hrf_model',
+        'standardize',
+        'high_pass',
+        'target_shape',
+        'signal_scaling',
+        'drift_order',
+        'scaling_axis',
+        'smoothing_fwhm',
+        'target_affine',
+        'slice_time_ref',
+        'fir_delays',
+    ]
+    # ATTRIBUTE_UNITS = {
+    #     't_r': 's',
+    #     'high_pass': 'Hz',
+    #     'fir_delays': 's',
+    # }
+    ATTRIBUTE_RENAMING = {
+        't_r': 'RepetitionTime',
+    }
+
+    # Fields for the top level of the dictionary
+    DATA_ATTRIBUTES.sort()
+    data_attributes = {
+        attr_name: getattr(model, attr_name)
+        for attr_name in PARAMETER_ATTRIBUTES
+        if hasattr(model, attr_name)
+    }
+    data_attributes = {
+        ATTRIBUTE_RENAMING.get(k, k): v for k, v in data_attributes.items()
+    }
+
+    # Fields for a nested section of the dictionary
+    # The ModelParameters field is an ad-hoc way to retain useful info.
+    PARAMETER_ATTRIBUTES.sort()
+    model_attributes = {
+        attr_name: getattr(model, attr_name)
+        for attr_name in PARAMETER_ATTRIBUTES
+        if hasattr(model, attr_name)
+    }
+    model_attributes = {
+        ATTRIBUTE_RENAMING.get(k, k): v for k, v in model_attributes.items()
+    }
+
+    model_metadata = {
+        'Description': 'A statistical map generated by nilearn.',
+        **data_attributes,
+        'ModelParameters': model_attributes,
+    }
+
+    dataset_description = {
+        'GeneratedBy': {
+            'Name': 'nilearn',
+            'Version': nilearn.__version__,
+            'Description': 'A nilearn {} GLM.'.format(
+                'first-level' if model_level == 1 else 'second-level'
+            ),
+            'CodeURL': (
+                'https://github.com/nilearn/nilearn/releases/tag/'
+                '{}'.format(nilearn.__version__)
+            )
+        }
+    }
+
+    with open(model_metadata_file, 'w') as fo:
+        json.dump(model_metadata, fo, indent=4, sort_keys=True)
+
+    with open(dataset_description_file, 'w') as fo:
+        json.dump(dataset_description, fo, indent=4, sort_keys=True)
+
+    # Write out statistical maps
+    for contrast_name, contrast_maps in statistical_maps.items():
+        contrast_name = _clean_contrast_name(contrast_name)
+
+        # Extract stat_type
+        # TODO: Figure out if this is correct
+        contrast_matrix = contrasts[contrast_name]
+        if contrast_matrix.ndim == 2:
+            stat_type = 'F'
+        else:
+            stat_type = 't'
```

Comment:
Is this a good way to determine the statistic type for the contrast? It's how contrast type is inferred when not explicitly set, and I don't think this function allows users to set contrast types explicitly, so it seems reasonable to also infer the statistic type the same way here, but I want to be sure.

Reply:
In  theory it is possible to one-dimensional F contrasts (that are equivalent to  two-sided t tests). To me this information has to be sued supplied. See e.g. the `contrast_type` in `compute_contrast` function

Reply:
Okay, thanks. Since the function accepts contrasts as a dictionary of name-array pairs, I'm not sure where to put the contrast type information. It could maybe be a separate parameter- a dictionary mapping contrast names to contrast types, with `None`s or missing values assumed to mean that automatic type detection (i.e., 1-dim = t, 2-dim = F) should be used. Does that sound reasonable?

Reply:
I have no better suggestion.

Reply:
I'll try the new parameter out then.